src/test/java/org/itsallcode/openfasttrace/importer/asciidoc/AsciiDocImporterTest.java

```diff
@@ -1,14 +1,175 @@
+                    ## A Requirement
+
+                    [.rationale]
+                    The rationale
+
+                    [.description]
+                    The description
+
+                    [.comment]
+                    A comment
+                    """,
+            """
+                    # Spec
+
+                    .A Requirement
+                    [.specitem, oft-sid="dsn~detail-design~1", oft-depends="dsn~grand-design~1, arch~general-constraints~1", oft-needs="impl, utest", oft-covers="req~first-requirement~1, req~second-requirement~1"]
+                    ====
+
+                    The description
+
+                    [.rationale]
+                    The rationale
+
+                    [.comment]
+                    A comment
+                    ====
+                    """,
+            """
+                    # Spec
+
+                    .A Requirement
+                    [.specitem, oft-sid="dsn~detail-design~1", oft-depends="dsn~grand-design~1, arch~general-constraints~1", oft-needs="impl, utest", oft-covers="req~first-requirement~1, req~second-requirement~1"]
+                    ====
+
+                    [.rationale]
+                    The rationale
+
+                    [.description]
+                    The description
+
+                    [.comment]
+                    A comment
+                    ====
+                    """
+    })
+    void testImporterReadsFullSpecItem(final String content)
+    {
+        final var importer = new AsciiDocImporter(content, listener);
+        importer.runImport();
+        inOrder.verify(listener).beginSpecificationItem();
+        Mockito.verify(listener).setId(new SpecificationItemId.Builder("dsn~detail-design~1").build());
+        Mockito.verify(listener).setLocation(any());
+        Mockito.verify(listener).setTitle("A Requirement");
+        Mockito.verify(listener).addDependsOnId(new SpecificationItemId.Builder("dsn~grand-design~1").build());
+        Mockito.verify(listener).addDependsOnId(new SpecificationItemId.Builder("arch~general-constraints~1").build());
+        Mockito.verify(listener).addNeededArtifactType("impl");
+        Mockito.verify(listener).addNeededArtifactType("utest");
+        Mockito.verify(listener).addCoveredId(new SpecificationItemId.Builder("req~first-requirement~1").build());
+        Mockito.verify(listener).addCoveredId(new SpecificationItemId.Builder("req~second-requirement~1").build());
+        Mockito.verify(listener).appendDescription("The description");
+        Mockito.verify(listener).appendRationale("The rationale");
+        Mockito.verify(listener).appendComment("A comment");
+        inOrder.verify(listener).endSpecificationItem();
+        Mockito.verifyNoMoreInteractions(listener);
+    }
```

Comment:
I was thinking, if you should test different section orders. But I dismissed that idea. You are using an external parser, so the parser itself isn't your unit under test. That means just testing different syntax variants that are supposed to yield the same result is sufficient.